### PR TITLE
INTG-2627 - Refactored code to allow for checking the db connection.

### PR DESCRIPTION
**Pull request by sc-gv**
Refactored code to allow for checking the db connection from the UI beforehand.

### CODE REVIEW

File: pkg/internal/feed/exporter_sql.go [hunk 1]
```diff
@@ -201,6 +201,21 @@ func (e *SQLExporter) WriteMedia(auditID, mediaID, contentType string, body []by
...
+	return &SQLExporter{
+		DB:              db,
+		Logger:          l,
+		AutoMigrate:     autoMigrate,
+		ExportMediaPath: exportMediaPath,
+		duration:        0,
+	}, nil
+}
+
+func GetDatabase(dialect string, connectionString string) (*zap.SugaredLogger, *gorm.DB, error) {
```
Comment (MickStanciu) [inline pkg/internal/feed/exporter_sql.go:218]: this function should not return a logger

File: pkg/api/exporter_api.go [hunk 2]
```diff
@@ -179,6 +179,14 @@ func (s *SafetyCultureExporter) RunInspectionJSON() error {
 	return nil
 }
 
+func (s *SafetyCultureExporter) CheckDBConnection() error {
```
Comment (MickStanciu) [inline pkg/api/exporter_api.go:182]: GoDoc :)

**Review by MickStanciu — CHANGES_REQUESTED** [review]
left a comment about logger

File: pkg/internal/feed/exporter_sql.go (hunk 1 above)
Comment (sc-gv) [reply pkg/internal/feed/exporter_sql.go]: yeah stupid me. goland refactoring isn't that smart after all. :P

File: pkg/api/exporter_api.go (hunk 2 above)
Comment (sc-gv) [reply pkg/api/exporter_api.go:182]: done

File: pkg/internal/feed/exporter_sql.go (hunk 1 above)
Comment (sc-gv) [reply pkg/internal/feed/exporter_sql.go]: done.

**Review by MickStanciu — APPROVED** [review]

**Comment by MickStanciu** [comment]
LGTM!!!

**Pull request merged**